tests/unit/update-manager-test.js

```diff
@@ -198,4 +197,53 @@ describe('UpdateManager', () => {
 
   })
 
+  describe('editable', () => {
+    const self = {err: ''}
+    let updater, docuri, rterm, options, userCallback, loadStub
+    var loadStatus = 200
+
+    beforeEach(() => {
+      options = {}
+      userCallback = () => {}
+
+      updater = new UpdateManager()
+      // updater.store.fetcher.webOperation = sinon.stub().resolves({ ok: true, status: 200, statusText: "Dummy stub 1"})
+      // updater.store.fetcher.load = sinon.stub().resolves({ ok: true, status: 200, statusText: "Dummy stub 2"})
+
+    })
+
+    it('Should detect a document is editable from metadata', () => {
+      loadMeta(updater.store)
+      expect(updater.editable(doc1)).to.equal('SPARQL')
+    })
+
+    it('Should not detect a document is editable from metadata after flush', () => {
+      loadMeta(updater.store)
+      updater.flagAuthorizationMetadata()
+      expect(updater.editable(doc1)).to.equal(undefined)
+    })
+
+
+    it('Async version should detect a document is editable from metadata', () => {
+      loadMeta(updater.store)
+      updater.checkEditable(doc1).then(
+        res => expect(result).to.equal('SPARQL')
+      )
+      expect(updater.editable(doc1)).to.equal('SPARQL')
+    })
+
+    it('Async version should not detect a document is editable from metadata after flush', () => {
+      loadMeta(updater.store)
+      updater.flagAuthorizationMetadata()
+      updater.checkEditable(doc1).then(
+        res => expect(result).to.equal('SPARQL')
```

Comment:
I think this for this to work we need to 1) include an accept-patch header in the (currently commented-out) mocks in the `beforeEach` and 2) make sure the fetcher stores the meta in the correct graph (as mentioned also in my other comment at https://github.com/linkeddata/rdflib.js/pull/512/files#r1137660733)